src/main/java/com/google/api/codegen/metacode/InitCodeGeneratorContext.java

```diff
@@ -43,6 +45,9 @@
 
   public abstract Method method();
 
+  @Nullable
+  public abstract ModelTypeTable typeTable();
+
```

Comment:
you still have a reference to ModelTypeTable in a metacode/* class here. All usage and knowledge should be confined to the transformer classes; metacode/* classes shouldn't be aware of this logic.